crates/ruff_linter/src/rules/fastapi/rules/fastapi_unused_path_parameter.rs

```diff
@@ -0,0 +1,198 @@
+        } = self;
+        if *arg_name_is_pos_only_arg {
+            format!(
+                "Path parameter `{arg_name}` in route path, but appears as a positional-only argument in function `{function_name}`. Consider making it a regular argument."
+            )
+        } else {
+            format!(
+                "Path parameter `{arg_name}` in route path but not in function `{function_name}`."
+            )
+        }
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        let Self { arg_name, .. } = self;
+        if self.arg_name_already_used {
+            None
+        } else {
+            Some(format!("Add `{arg_name}` to function signature"))
+        }
+    }
+}
+
+/// Returns a vector of path parameters and their ranges in the route path.
+/// The string is just the name of the path parameter.
+/// The range includes the curly braces.    
+fn extract_path_params_from_route(input: &str) -> Vec<(String, Range<usize>)> {
+    // We ignore text after a colon, since those are path convertors
+    // See also: https://fastapi.tiangolo.com/tutorial/path-params/?h=path#path-convertor
+    let re = Regex::new(r"\{([^:}]+).*?\}").unwrap();
+
+    // Collect all matches and return them as a vector of strings
+    re.captures_iter(input)
+        .filter_map(|cap| {
+            if let (Some(name_match), Some(full_match)) = (cap.get(1), cap.get(0)) {
+                Some((name_match.as_str().trim().to_string(), full_match.range()))
+            } else {
+                None
+            }
+        })
+        .collect()
+}
+
+pub(crate) fn fastapi_unused_path_parameter(
+    checker: &mut Checker,
+    function_def: &ast::StmtFunctionDef,
+) {
+    if !checker.semantic().seen_module(Modules::FASTAPI) {
+        return;
+    }
+    if !is_fastapi_route(function_def, checker.semantic()) {
+        return;
+    }
+
+    // Get the route path from the decorator
+    let route_decorator = function_def
+        .decorator_list
+        .iter()
+        .find_map(|decorator| is_fastapi_route_decorator(decorator, checker.semantic()));
+
+    let Some(route_decorator) = route_decorator else {
+        return;
+    };
+
+    let path_arg = route_decorator.arguments.args.first();
+    let Some(path_arg) = path_arg else {
+        return;
+    };
+    // Lets path_arg go out of scope so we can reuse checker later
+    let diagnostic_range = path_arg.range();
+    // We can't really handle anything other than string literals
+    let path = match path_arg.as_string_literal_expr() {
+        Some(path_arg) => path_arg.value.to_string(),
+        None => return,
+    };
+
+    let path_params = extract_path_params_from_route(&path);
+
+    // Now we extract the arguments from the function signature
+    let named_args = function_def
+        .parameters
+        .args
+        .iter()
+        .chain(function_def.parameters.kwonlyargs.iter())
+        .map(|arg| arg.parameter.name.to_string())
```

Comment:
I think we can avoid calling `to_string` here and instead just use the borrowed `&str`
```suggestion
        .map(|arg| &arg.parameter.name)
```